cobbler/settings/migrations/V3_0_0.py

```diff
@@ -223,5 +227,142 @@ def migrate(settings: dict) -> dict:
     for key in deleted_keys:
         helper.key_delete(key, settings)
 
-    # TODO: v2 to v3 script
+    # START: migrate-data-v2-to-v3
+    def serialize_item(collection, item):
+        """
+        Save a collection item to file system
+
+        :param collection: name
+        :param item: dictionary
+        """
+        filename = "/var/lib/cobbler/collections/%s/%s" % (collection, item["name"])
+
+        if settings.get("serializer_pretty_json", False):
+            sort_keys = True
+            indent = 4
+        else:
+            sort_keys = False
+            indent = None
+
+        filename += ".json"
+        with open(filename, "w+") as item_fd:
+            data = json.dumps(
+                item, encoding="utf-8", sort_keys=sort_keys, indent=indent
+            )
+            item_fd.write(data)
+
+    def deserialize_raw_old(collection_types):
+        results = []
+        all_files = glob.glob("/var/lib/cobbler/config/%s/*" % collection_types)
+
+        for file in all_files:
+            with open(file) as item_fd:
+                json_data = item_fd.read()
+                _dict = json.loads(json_data, encoding="utf-8")
+                results.append(_dict)
+        return results
+
+    def substitute_paths(value):
+        if isinstance(value, list):
+            new_value = []
+            for item in value:
+                new_value.append(substitute_paths(item))
+            value = new_value
+        elif isinstance(value, str):
+            value = value.replace("/ks_mirror/", "/distro_mirror/")
+        return value
+
+    def transform_key(key, value):
+        if key in transform:
+            ret_value = transform[key](value)
+        else:
+            ret_value = value
```

Comment:
Alternative, not sure if it's better
```
def transform_key(key, value):
    return substitute_paths(transform.get(key, lambda value: value)(value))
```

Reply:
I am having a hard time reading that. I think I will stick with what we have.